### DISCUSSION TITLE
[REQUEST] Documentation: Write a tutorial to use Helix as an IDE

### DISCUSSION THREAD

**Discussion by mydoghasfleas** [helix-editor/helix#9561] [Ideas]
Most of the documentation seems to assume some familiarity with a vi-type editor like Vim, or are focused on the many commands for editing text.

It would be nice to have a very practical tutorial showing how to use Helix as an IDE, whereby the user can pick up the necessary skills along the way as they go.

It would be helpful to have topics like: how to edit files in a directory/project structure, how to use editor features like auto-complete, how to compile source code, etc.

Maybe pick a language, like Rust, but provide some general concepts that can easily be extended to other languages.

It should be aimed at people coming from other IDEs, assuming they are newcomers to TUI editors, but showing how they would accomplish the same tasks.

**Why `:tutor` doesn't cut it:**

`:tutor` (which is anyway still a work in progress) is focused on all the functionality available for editing and navigating text. It does not (as yet) cover topics such as dealing with multiple files (projects) and is decidedly not _task-oriented_, like using Helix as an IDE. It is more of a massive information dump, leading to cognitive overload.

To use an analogy, I think `:tutor` is more like learning the piano by beginning with music theory, practicing scales, reading music, etc. Lots of information, very slow to get going.

The tutorial I am hoping someone can write would be more akin to learning piano chords to accompany songs you already know, allowing you to feel productive much quicker. You can still pick up all the music theory as you go along.

**Comment by TornaxO7** [comment]
May I ask why `:tutor` doesn't fullfill this?

**Comment by mydoghasfleas** [comment]
Good question, and probably worth adding into the main requirement!
Let me update my initial request to include that information.

**Comment by archseer** [comment]
In my opinion this is better suited as external documentation or a blog article. An "IDE" workflow is specific to a single language, and the editing experience will vary widely based on the language server used.

**Comment by quantonganh** [comment]
@mydoghasfleas Here's my note: https://quantonganh.com/posts/2023/08/19/turn-helix-into-ide/

**Reply by mydoghasfleas** [reply]
Thank you! Some nice tips on there.  I've just discovered Zellij but will check out Wezterm 😀

**Reply by tomsturges** [reply]
file not found

**Reply by quantonganh** [reply]
@tomsturges I just moved my blog to GitHub Pages, and the link has changed to https://quantonganh.com/posts/2023/08/19/turn-helix-into-ide/.

**Comment by tomsturges** [comment]
Thanks :) also the links in that page are broken too
